src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.OSX.cs

```diff
@@ -390,5 +434,61 @@ public static SecurityStatusPal ApplyShutdownToken(
                 SecurityStatusPalErrorCode.InternalError,
                 Interop.AppleCrypto.CreateExceptionForOSStatus(osStatus));
         }
+
+        internal static bool ShouldUseAsyncSecurityContext(SslAuthenticationOptions sslAuthenticationOptions)
+        {
+            return ShouldUseNetworkFramework(sslAuthenticationOptions);
+        }
+
+        private static bool ShouldUseNetworkFramework(
+            SslAuthenticationOptions sslAuthenticationOptions)
+        {
+            return
+                sslAuthenticationOptions.IsClient &&
+                SafeDeleteNwContext.IsNetworkFrameworkAvailable &&
+                (sslAuthenticationOptions.EnabledSslProtocols == SslProtocols.None ||
+                   sslAuthenticationOptions.EnabledSslProtocols == SslProtocols.Tls13 ||
+                   sslAuthenticationOptions.EnabledSslProtocols == SslProtocols.Tls13 ||
```

Comment:
Duplicate line? I would think this should be true as long as `Tls13` is in the set regardless of other protocol combinations.